src/main/java/io/spokestack/spokestack/tts/TTSCallback.java

```diff
@@ -72,6 +95,38 @@ protected AudioResponse createAudioResponse(
         return new AudioResponse(metadata, audioUri);
     }
 
+    private void deliverResponse(String requestId, AudioResponse response) {
+        List<QueuedCall> newQueue = new ArrayList<>();
+        boolean dispatch = true;
+        // iterate through the queue each time a response is received to make
+        // sure they're delivered in the order they were submitted
+        for (QueuedCall queuedCall : this.callQueue) {
```

Comment:
this callback class is now stateful, and it looks like okhttp dispatches responses via a thread pool, so you may need to protect your state from concurrent access

Reply:
Ah, good catch. Thanks.